api/envoy/extensions/filters/http/proto_api_scrubber/v3/config.proto

```diff
@@ -0,0 +1,323 @@
+//               }
+//             }
+//           },
+//           "book.debug_info": {
+//             "restrictions":{
+//               "USER_TYPE": {
+//                 "any": {
+//                   "restrictions": [{"value": "DEV"}]
+//                 }
+//               }
+//             }
+//           }
+//         }
+//       }
+//     }
+//   }
+// }
+//
+// Request and Response
+// --------------------
+//
+// Consider the following request headers and body received by the filter for
+// the BookService.GetBook method:
+//
+// Request Headers
+//
+// .. code-block:: json
+//
+// {
+//   "header1": "value1",
+//   "header2": "value2",
+//   "X-USER-TYPE": "PROD"
+// }
+//
+// Request Body
+//
+// .. code-block:: json
+//
+// {
+//   "book_id": ABC1234
+// }
+//
+// And consider the following response body received by the filter corresponding
+// to the above request:
+//
+// .. code-block:: json
+//
+// {
+//   "book": {
+//     "title": "Book Title",
+//     "author": "Book Author",
+//     "publisher": "Book Publisher",
+//     "debug_info": "This books metadata is stored in database shard - 0004"
+//   },
+//   "debug_info": "Served from server with IP: 172.164.1.2"
+// }
+//
+// The filtered response output by this filter will be the following:
+//
+// .. code-block:: json
+//
+// {
+//   "book": {
+//     "title": "Book Title",
+//     "author": "Book Author",
+//     "publisher": "Book Publisher"
+//   }
+// }
+//
+// Note that the fields `debug_info` and `book.debug_info` are filtered out from
+// the response since the request entitlement of type "USER_TYPE" is PROD while
+// the restrictions on the fields `debug_info` and `book.debug_info` is DEV.
+
+// [#next-free-field: 6]
+message ProtoApiScrubberConfig {
+  // Specifies the filtering mode of this filter.
+  // Currently only ``OVERRIDE`` mode is supported.
+  enum FilteringMode {
+    // If unspecified, falls back to ``OVERRIDE``.
+    FILTERING_MODE_UNSPECIFIED = 0;
+
+    // Override the original request/response body with the filtered
+    // request/response body.
+    OVERRIDE = 1;
+  }
+
+  // The proto descriptor set for the proto services.
+  oneof descriptor_set {
+    // It could be passed by a local file through ``Datasource.filename`` or
+    // embedded in the ``Datasource.inline_bytes``.
+    config.core.v3.DataSource data_source = 1;
+
+    // Unimplemented, the key of proto descriptor TypedMetadata.
+    // Among filters depending on the proto descriptor, we can have a
+    // TypedMetadata for proto descriptors, so that these filters can share one
+    // copy of proto descriptor in memory.
+    string proto_descriptor_typed_metadata = 2;
+  }
+
+  // Data sources for the request entitlements.
+  // Entitlements are matched with restrictions of the same name at runtime.
+  // Make sure that the entitlement source is present for each of the
+  // corresponding restrictions.
+  // Key - entitlement name
+  // Value - entitlement source
```

Comment:
```suggestion
  // - Key - entitlement name
  // - Value - entitlement source
```
not sure if any/all of Key/Value/name/source are "literals" - if so they should be double backticked - im guessing its just `name` and `source` that are literals